x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -610,8 +615,8 @@ private MethodSpec addIntermediateInput() {
                     ).map(Methods::requireType).toArray(TypeMatcher[]::new)
                 )
             );
-            if (intermediateState.stream().map(IntermediateStateDesc::elementType).anyMatch(n -> n.equals("BYTES_REF"))) {
-                builder.addStatement("$T scratch = new $T()", BYTES_REF, BYTES_REF);
+            for (IntermediateStateDesc interState : intermediateState) {
+                interState.addScratchDeclaration(builder);
```

Comment:
I think this was buggy prior to my change, but without causing a defect?

If there were multiple `BYTES_REF` state-members, they would have shared the same scratch, which seems incorrect?

Reply:
That seems possible. I imagine we don't have any intermediate states with two strings.